packages/react-components/react-jsx-runtime/src/createElement.ts

```diff
@@ -1,38 +1,60 @@
 import * as React from 'react';
-import { SlotRenderFunction, UnknownSlotProps, SLOT_RENDER_FUNCTION_SYMBOL } from '@fluentui/react-utilities';
-
-type WithMetadata<Props extends {}> = Props & {
-  [SLOT_RENDER_FUNCTION_SYMBOL]: SlotRenderFunction<Props>;
-};
+import {
+  UnknownSlotProps,
+  isSlot,
+  SlotComponent,
+  SLOT_COMPONENT_METADATA_SYMBOL,
+  slot,
+} from '@fluentui/react-utilities';
 
 export function createElement<P extends {}>(
   type: React.ElementType<P>,
   props?: P | null,
   ...children: React.ReactNode[]
 ): React.ReactElement<P> | null {
-  return hasRenderFunction(props)
-    ? createElementFromRenderFunction(type, props, children)
-    : React.createElement(type, props, ...children);
+  // TODO:
+  // this is for backwards compatibility with getSlotsNext
+  // it should be removed once getSlotsNext is obsolete
+  if (isSlot<P>(props)) {
+    return createElementFromSlotComponent(
```

Comment:
Let's make this somehow differently as line 24 has the same condition

Reply:
is the same method with a different argument... they're not equivalent, one is checking for old implementation (where the symbols will be provided through `props`) and the one in line 24 is checking for the new one (where the symbols will be provided through `type`)

Reply:
Ouch, I see. Sorry for confusion